activerecord/test/cases/adapters/postgresql/uuid_test.rb

```diff
@@ -403,3 +403,41 @@ def test_find_by_with_uuid
     assert_nil UuidPost.find_by(id: 789)
   end
 end
+
+class PostgresqlUUIDCaseInsensitive < ActiveRecord::PostgreSQLTestCase
+  include PostgresqlUUIDHelper
+
+  class UuidModel < ActiveRecord::Base
+    self.table_name = "pg_uuid_models"
+  end
+
+  setup do
+    connection.transaction do
+      connection.create_table("pg_uuid_models", id: :uuid, **uuid_default) do |t|
+        t.string "title"
+      end
+    end
+  end
+
+  teardown do
+    drop_table "pg_uuid_models"
+  end
+
+  def test_upcase_not_dirty
+    model = UuidModel.create!(id: "abcd-0123-4567-89ef-dead-beef-0101-1010")
+    model.id = "ABCD-0123-4567-89EF-DEAD-BEEF-0101-1010"
+    assert_not model.changed?
```

Comment:
I think a test like this might be easier to read / understand if the UUID is pulled out into a variable:
```ruby
    uuid = "abcd-0123-4567-89ef-dead-beef-0101-1010"
    model = UuidModel.create!(id: uuid)
    model.id = uuid.swapcase
    assert_not model.changed?
```

Reply:
I made this change and the corresponding change in the MAC address tests, too.